homeassistant/components/smartthings/smartapp.py

```diff
@@ -176,6 +175,7 @@ def setup_smartapp(hass, app):
         webhook.async_generate_path(config[CONF_WEBHOOK_ID]),
         dispatcher=dispatcher)
     manager.connect_install(functools.partial(smartapp_install, hass))
+    manager.connect_update(functools.partial(smartapp_update, hass))
```

Comment:
Will this require reconnecting the smart app ?

Reply:
This change actually makes it so the user doesn't have to reinstall the app to get the new functionality--they still have to 'update' the app in SmartThings (go into the HASS app and tap 'Done').  

I want to make this unnecessary (and synchronize the subscriptions when the config entry is loaded), but there's a bug in the SmartThings API that I have to figure out a workaround for.